### DISCUSSION TITLE
Set all blocks in an instance container into air

### DISCUSSION THREAD

**Discussion by Codestech1** [Minestom/Minestom#1855] [Help] [answered]
Hi, how can I set all blocks in an instance container into air?

**Comment by emortaldev** [comment]
For what purpose?
It might just be easier to create a new instance without any loaders or generators depending on what you want to do.

**Reply by Codestech1** [reply]
I am coding minigame right now and i don't want to create new instance on game restart, i just want to clear all blocks to don't make trash in the cache.

**Comment by Codestech1** [comment, marked as answer]
Again, found it by myself
```
instance.chunks.forEach {
    it.sections.forEach { sec ->
        sec.blockPalette().getAllPresent { x, y, z, value ->
            if (Block.fromStateId(value.toShort()) != Block.AIR) {
                gameInstance.setBlock(x, y, z, Block.AIR)
            }
        }
    }
}
```
